ts/Core/Tooltip.ts

```diff
@@ -1068,15 +1068,16 @@ class Tooltip {
                         });
                     }
 
+                    // Set the stroke color of the box to reflect the point
+                    // Add class before the label BBox is calculated (#21035)
+                    label.addClass(tooltip.getClassName(point), true);
```

Comment:
The `addClass` function is chainable, so we can save some bytes by doing

`label.addClass()
    .attr();`

Alternatively, I think we can just pass the class as an attribute. Though I am not sure if the required order is preserved. Can you check if this works?

```ts
                    label.attr({
                        // Add class before the label BBox is calculated (#21035)
                        'class': tooltip.getClassName(point),
                        text: text && (text as any).join ?
                            (text as any).join('') :
                            text
                    });
```

Reply:
Passing class as an attribute does the work and the order is preserved correctly, so I'd opt to go with this approach.